portal/src/components/roles-and-groups/dialog/AddUserRolesDialog.tsx

```diff
@@ -68,7 +68,12 @@ export const AddUserRolesDialog: React.VFC<AddUserRolesDialogProps> =
         const selectedRoleIDs = new Set(
           selectedTags?.map((tag) => (tag as RoleTag).role.id)
         );
-        const result = await refetch({ searchKeyword: filter, pageSize: 100 });
+        const excludedIDs = new Set([...selectedRoleIDs, ...existingRoleIDs]);
+        const result = await refetch({
+          searchKeyword: filter,
+          excludedIDs: [...excludedIDs],
```

Comment:
Is being able to select a role already assigned to user due to group expected?

Reply:
Yes, it doesn't have such restriction